mwclient/client.py

```diff
@@ -979,7 +993,16 @@ def ask(self, query, title=None):
         while offset is not None:
             results = self.raw_api('ask', query='{query}|offset={offset}'.format(
                 query=query, offset=offset), http_method='GET', **kwargs)
-
-            offset = results.get('query-continue-offset')
-            for key, value in results['query']['results'].iteritems():
-                yield {key: value}
+            if results.get('error'):
+                if self.handle_api_result(results):
+                    offset = None
+                    yield {}
+            elif results.get('query'):
+                answer = results.get('query').get('results')
+                if (answer):
```

Comment:
If the query is valid but there are no results, answer is an empty list,  so 
- if answer -> False
but
- if answer is not None  -> True

We should stick to "if answer"